### PR TITLE
Code cleanups

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

File: moshi/src/main/java/com/squareup/moshi/JsonUtf8Writer.java [hunk 1]
```diff
@@ -358,15 +358,14 @@ public void close() throws IOException {
    * and escapes those characters that require it.
    */
   static void string(BufferedSink sink, String value) throws IOException {
-    String[] replacements = REPLACEMENT_CHARS;
```
Comment (JakeWharton) [inline moshi/src/main/java/com/squareup/moshi/JsonUtf8Writer.java:361]: This is a performance enhancement in very, very inner-loop code which caches the static field lookup in a local. Please revert.

File: adapters/src/main/java/com/squareup/moshi/adapters/PolymorphicJsonAdapterFactory.java [hunk 2]
```diff
@@ -212,8 +212,8 @@ public void toJson(JsonWriter writer, Object value) throws IOException {
     }
 
     List<JsonAdapter<Object>> jsonAdapters = new ArrayList<>(subtypes.size());
-    for (int i = 0, size = subtypes.size(); i < size; i++) {
-      jsonAdapters.add(moshi.adapter(subtypes.get(i)));
+    for (Type subtype : subtypes) {
+      jsonAdapters.add(moshi.adapter(subtype));
```
Comment (JakeWharton) [inline adapters/src/main/java/com/squareup/moshi/adapters/PolymorphicJsonAdapterFactory.java:216]: Please revert. Indexed-based loops do not allocate.

File: moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java [hunk 3]
```diff
@@ -313,8 +313,7 @@ public Object fromJson(Moshi moshi, JsonReader reader)
   /** Returns the matching adapter method from the list. */
   private static @Nullable AdapterMethod get(
       List<AdapterMethod> adapterMethods, Type type, Set<? extends Annotation> annotations) {
-    for (int i = 0, size = adapterMethods.size(); i < size; i++) {
-      AdapterMethod adapterMethod = adapterMethods.get(i);
+    for (AdapterMethod adapterMethod : adapterMethods) {
```
Comment (JakeWharton) [inline moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java:316]: Please revert. Indexed-based loops do not allocate.

File: moshi/src/main/java/com/squareup/moshi/Moshi.java [hunk 4]
```diff
@@ -141,8 +141,8 @@
       if (adapterFromCall != null) return adapterFromCall;
 
       // Ask each factory to create the JSON adapter.
-      for (int i = 0, size = factories.size(); i < size; i++) {
-        JsonAdapter<T> result = (JsonAdapter<T>) factories.get(i).create(type, annotations, this);
+      for (JsonAdapter.Factory factory : factories) {
+        JsonAdapter<T> result = (JsonAdapter<T>) factory.create(type, annotations, this);
```
Comment (JakeWharton) [inline moshi/src/main/java/com/squareup/moshi/Moshi.java:145]: Please revert. Indexed-based loops do not allocate.

File: moshi/src/main/java/com/squareup/moshi/Moshi.java [hunk 5]
```diff
@@ -321,8 +321,7 @@ public Moshi build() {
      */
     <T> JsonAdapter<T> push(Type type, @Nullable String fieldName, Object cacheKey) {
       // Try to find a lookup with the same key for the same call.
-      for (int i = 0, size = callLookups.size(); i < size; i++) {
-        Lookup<?> lookup = callLookups.get(i);
+      for (Lookup<?> lookup : callLookups) {
```
Comment (JakeWharton) [inline moshi/src/main/java/com/squareup/moshi/Moshi.java:324]: Please revert. Indexed-based loops do not allocate.

File: moshi/src/main/java/com/squareup/moshi/Moshi.java [hunk 6]
```diff
@@ -356,8 +355,7 @@ void pop(boolean success) {
 
       if (success) {
         synchronized (adapterCache) {
-          for (int i = 0, size = callLookups.size(); i < size; i++) {
-            Lookup<?> lookup = callLookups.get(i);
+          for (Lookup<?> lookup : callLookups) {
```
Comment (JakeWharton) [inline moshi/src/main/java/com/squareup/moshi/Moshi.java:358]: Please revert. Indexed-based loops do not allocate.

File: moshi/src/main/java/com/squareup/moshi/LinkedHashTreeMap.java [hunk 7]
```diff
@@ -346,13 +346,11 @@ private void rebalance(Node<K, V> unbalanced, boolean insert) {
...
         int rightDelta = rightLeftHeight - rightRightHeight;
-        if (rightDelta == -1 || (rightDelta == 0 && !insert)) {
-          rotateLeft(node); // AVL right right
-        } else {
+        if (rightDelta != -1 && (rightDelta != 0 || insert)) {
           assert (rightDelta == 1);
           rotateRight(right); // AVL right left
-          rotateLeft(node);
         }
+        rotateLeft(node); // AVL right right
```
Comment (swankjesse) [inline moshi/src/main/java/com/squareup/moshi/LinkedHashTreeMap.java:353]: These comments no longer make sense. This `right right` comment is now being applied to the right-left case.

File: moshi/src/main/java/com/squareup/moshi/Moshi.java [hunk 8]
```diff
@@ -184,7 +184,7 @@
   @CheckReturnValue
   public Moshi.Builder newBuilder() {
     Builder result = new Builder();
-    for (int i = 0, limit = lastOffset; i < limit; i++) {
+    for (int i = 0; i < lastOffset; i++) {
```
Comment (swankjesse) [inline moshi/src/main/java/com/squareup/moshi/Moshi.java:187]: This replaces a local read with a field read. Probably not a meaningful performance regression but in general we like optimized loops that don’t have to repeatedly read memory when they don’t have to.

**Pull request merged**